src/unix/darwin.c

```diff
@@ -333,3 +333,29 @@ void uv_free_interface_addresses(uv_interface_address_t* addresses,
 
   uv__free(addresses);
 }
+
+int uv__tcp_enable_largesocket(uv_tcp_t *tcp) {
+  int rmem;
+  int wmem;
+  int r;
+
+  size_t size = sizeof(int);
+  if (sysctlbyname("net.inet.tcp.recvspace", &rmem, &size, NULL, 0) ||
+      rmem <= STDTCPWINDOW)
+    return UV_ENOSYS;
+  
+  if (sysctlbyname("net.inet.tcp.sendspace", &wmem, &size, NULL, 0) ||
```

Comment:
> The CI bot is freebasd only, so I wasn't able to test for other BSD variants.

I have a OpenBSD VM somewhere, I'll try to run the tests from your branch and report back.

> There is a mention about dragonfly BSD variant in the documentation, and usage of it in libuv code. I wasn't sure about its relevance in this context, so omitted it for the time. What do you suggest?

I think `DragonFly BSD` should work with your changes too, so I would add the support. See: https://leaf.dragonflybsd.org/cgi/web-man?command=tuning&section=7 and https://leaf.dragonflybsd.org/cgi/web-man?command=sysctl&section=3